.github/workflows/release.yml

```diff
@@ -1,86 +1,100 @@
 name: Release version
+
 on:
   push:
     tags:
       - 'v*.*.*'
+  schedule:
+    - cron: '0 0 * * *'
+
+env:
+  TAG_NAME: ${{ github.event_name == 'schedule' && 'nightly' || github.ref_name }}
+  IS_NIGHTLY: ${{ github.event_name == 'schedule' }}
+
 jobs:
-  # this job builds the `forge` and `cast` binaries for the specified platforms
-  # in the matrix. The MacOS binaries do not support M1 yet: https://github.com/actions/runner/issues/805
-  build-artifacts:
-    runs-on:  ${{ matrix.os }}
+  release:
+    name: ${{ matrix.job.target }} (${{ matrix.job.os }})
+    runs-on: ${{ matrix.job.os }}
     strategy:
       matrix:
-        os: [macos-latest, ubuntu-latest]
+        job:
+          # The OS is used for the runner
+          # The platform is a generic platform name
+          # The target is used by Cargo
+          # The arch is either 386 or amd64
+          - os: ubuntu-latest
+            platform: linux
+            target: x86_64-unknown-linux-gnu
+            arch: amd64
+          # The macOS binaries do not support M1 yet: https://github.com/actions/runner/issues/805
+          - os: macos-latest
+            platform: darwin
+            target: x86_64-apple-darwin
+            arch: amd64
+          - os: windows-latest
+            platform: win32
+            target: x86_64-pc-windows-msvc
+            arch: amd64
+
     steps:
       - name: Checkout sources
         uses: actions/checkout@v2
+
       - name: Install toolchain
         uses: actions-rs/toolchain@v1
         with:
           profile: minimal
           toolchain: stable
           override: true
+
       - uses: Swatinem/rust-cache@v1
         with:
           cache-on-failure: true
-      - name: cargo build
+
+      - name: Build binaries
         uses: actions-rs/cargo@v1
         with:
           command: build
-          args: --release
-
-      - name: Upload production artifacts
-        uses: actions/upload-artifact@v2
-        with:
-            name: target-${{ matrix.os }}
-            path: |
-              ./target/release/forge
-              ./target/release/cast
+          args: --release --bins
 
-  # creates a GH release with the built binaries and CHANGELOG entry
-  create-release:
-    runs-on: ubuntu-latest
-    needs:  build-artifacts
-    steps:
-     - name: Checkout
-       uses: actions/checkout@v2
-       with:
-        fetch-depth: 1000
-
-     - name: Set output
-       id: vars
-       run: echo ::set-output name=tag::${GITHUB_REF#refs/*/}
-
-     - name: Restore artifacts
-       uses: actions/download-artifact@v2
+      - name: Strip binaries (non-Windows)
+        if: matrix.job.platform != 'win32'
+        run: |
+          cd target/release
+          strip cast
+          strip forge
```

Comment:
Not sure what the cost of this is for us while we are still moving fast, but it does reduce the footprint of both `forge` and `cast` significantly - cast is reduced to 6 megabytes (from 66 megs) and forge is similarly reduced from 106 megabytes to 9.8 megabytes

I noticed that `strip` can also reduce the footprint of Windows executables but it is not available in the workflow runner so :man_shrugging:

Reply:
No cost really, `strip` is 100% desired. 

more ideas here https://github.com/johnthagen/min-sized-rust

Reply:
Added some more of them